src/python/ensembl/io/genomio/events_loader.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from pathlib import Path
+from typing import Dict, List
+from dataclasses import dataclass
+
+import argschema
+from sqlalchemy.orm import Session
+
+from ensembl.database import DBConnection
+from ensembl.core.models import MappingSession, StableIdEvent
+
+
+@dataclass
+class IdEvent:
+    """Simple representation for the events from the input file"""
+    from_id: str
+    to_id: str
+    release: str
+    release_date: str
+
+
+class MapSession:
+    """Simple mapping_sessions representation from the input file"""
+    def __init__(self, release: str, release_date: str) -> None:
+        self.release = release
+        self.release_date = release_date
+        self.events: List[IdEvent] = []
+
+    def add_event(self, event: IdEvent) -> None:
+        """Add an event to this mapping_session"""
+        self.events.append(event)
+
+
+def load_events(input_file: Path) -> List[IdEvent]:
+    """Load events from input file.
+    Expected tab file columns: old_id, new_id, event_name, release, release_date
+    
+    """
+    events: List[IdEvent] = []
+
+    with input_file.open("r") as events_fh:
+        for line in events_fh:
+            line.strip()
+            if line == "":
+                continue
+            (from_id, to_id, _, release, release_date) = line.split("\t")
+            event = IdEvent(from_id=from_id, to_id=to_id, release=release, release_date=release_date)
+            events.append(event)
+    return events
+
+
+def write_events(session: Session, events: List[IdEvent], update: bool = False) -> None:
+    """Insert the events in the core database.
+    A mapping session is created for each different 'release'.
+    
+    """
+    # First, create mapping_sessions based on the release
+    mappings: Dict[str, MapSession] = {}
+    for event in events:
+        release = event.release
+
+        if release in mappings:
+            mappings[release].add_event(event)
+        else:
+            mappings[release] = MapSession(release, event.release_date)
+
+    # Then, add the mapping, and the events for this mapping
+    for release, mapping in mappings.items():
+        print(f"Adding mapping for release {release} ({len(mapping.events)} events)")
```

Comment:
Just a more generic question: should we consider moving from `print` messages to a proper log file?

Reply:
Yes we should...